### PR TITLE
[FEAT] Mine gold rock 

### PR DESCRIPTION
**Pull request by HBandesh**
# Description

In this PR, We have added the functionality to mine gold rocks in the land expansion. 

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/events/landExpansion/mineGold.test.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+      } as LandExpansionMineGoldAction,
+    };
+
+    const game = mineGold(payload);
+
+    expect(game.inventory["Iron Pickaxe"]).toEqual(new Decimal(0));
+    expect(game.inventory.Gold).toEqual(new Decimal(2));
+  });
+
+  it("mines multiple iron", () => {
```
Comment (mGranTY) [inline src/features/game/events/landExpansion/mineGold.test.ts:172]: should be `it("mines multiple gold", () => {`

File: src/features/game/expansion/components/resources/Gold.tsx [hunk 2]
```diff
@@ -1,16 +1,338 @@
...
+export const Gold: React.FC<Props> = ({ rockIndex, expansionIndex }) => {
+  const { gameService, selectedItem } = useContext(Context);
+  const [game] = useActor(gameService);
+
+  const [showPopover, setShowPopover] = useState(true);
+  const [errorLabel, setErrorLabel] = useState<"noStamina" | "noPickaxe">();
+  const [popover, setPopover] = useState<JSX.Element | null>();
+
+  const [touchCount, setTouchCount] = useState(0);
+  // When to hide the wood that pops out
```
Comment (mGranTY) [inline src/features/game/expansion/components/resources/Gold.tsx:53]: i think this comment is wrong for all resources(gold, iron, stone...).

File: src/features/game/expansion/components/resources/Gold.tsx [hunk 3]
```diff
@@ -1,16 +1,338 @@
...
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const sparkGif = useRef<SpriteSheetInstance>();
+  const minedGif = useRef<SpriteSheetInstance>();
+
+  const [showRockTimeLeft, setShowRockTimeLeft] = useState(false);
+
+  const { setToast } = useContext(ToastContext);
+  const expansion = game.context.state.expansions[expansionIndex];
+  const rock = expansion.gold?.[rockIndex] as LandExpansionRock;
```
Comment (mGranTY) [inline src/features/game/expansion/components/resources/Gold.tsx:65]: Iron.ts uses IronRock instead of just rock, we should do GoldRock here.

**Review by adamhannigan — DISMISSED** [review]
LGTM 🚀 Some nice suggestions from @mGranTY 😄

**Review by Rofrtd — APPROVED** [review]
LGTM

**Pull request merged**